renku/domain_model/project.py

```diff
@@ -36,6 +36,10 @@ class Project(persistent.Persistent):
 
     keywords: List[str] = list()
 
+    # NOTE: the minimum version of renku to needed to work with a project
+    # This should be bumped on metadata version changes and when we do non forward-compatible on-the-fly migrations
+    minimum_renku_version: str = "1.2.0"
```

Comment:
I suggest we put this in `renku.version` so that all version-related constants are in the same place.